File: frontend/lib/models/daily_cash_entry.dart
class DailyCashEntry {
  final int id;
  final int showroomId;
  final String? showroomName;
  final int userId;
  final String? userName;
  final String entryDate;
  final double cashAmount;
  final String? notes;
  final bool isLocked;
  final String cashAccountType;
  final String cashAccountLabel;
  final String? createdAt;
  final String? updatedAt;

  DailyCashEntry({
    required this.id,
    required this.showroomId,
    this.showroomName,
    required this.userId,
    this.userName,
    required this.entryDate,
    required this.cashAmount,
    this.notes,
    required this.isLocked,
    this.cashAccountType = 'main',
    this.cashAccountLabel = 'Main Account',
    this.createdAt,
    this.updatedAt,
  });

  factory DailyCashEntry.fromJson(Map<String, dynamic> json) {
    final accountType = (json['cash_account_type'] as String?) ?? 'main';
    return DailyCashEntry(
      id: json['id'] as int,
      showroomId: json['showroom_id'] as int,
      showroomName: json['showroom']?['name'] as String?,
      userId: json['user_id'] as int,
      userName: json['user']?['name'] as String?,
      entryDate: json['entry_date'] as String,
      cashAmount: double.tryParse(json['cash_amount'].toString()) ?? 0.0,
      notes: json['notes'] as String?,
      isLocked: json['is_locked'] == true || json['is_locked'] == 1,
      cashAccountType: accountType,
      cashAccountLabel: (json['cash_account_label'] as String?) ??
          (accountType == 'mano' ? "Mano's Account" : 'Main Account'),
      createdAt: json['created_at'] as String?,
      updatedAt: json['updated_at'] as String?,
    );
  }
}

File: ios/Runner/Models/DailyCashEntry.swift
import Foundation

struct DailyCashEntry: Decodable, Identifiable {
    let id: Int
    let showroomId: Int
    let showroomName: String?
    let userId: Int
    let userName: String?
    let entryDate: String
    let cashAmount: Double
    let notes: String?
    let isLocked: Bool
    let cashAccountType: String
    let cashAccountLabel: String
    let createdAt: String?
    let updatedAt: String?

    private enum CodingKeys: String, CodingKey {
        case id
        case showroomId        = "showroom_id"
        case showroom
        case userId            = "user_id"
        case user
        case entryDate         = "entry_date"
        case cashAmount        = "cash_amount"
        case notes
        case isLocked          = "is_locked"
        case cashAccountType   = "cash_account_type"
        case cashAccountLabel  = "cash_account_label"
        case createdAt         = "created_at"
        case updatedAt         = "updated_at"
    }

    private struct NameNested: Decodable { let name: String }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id          = try c.decode(Int.self, forKey: .id)
        showroomId  = try c.decode(Int.self, forKey: .showroomId)
        userId      = try c.decode(Int.self, forKey: .userId)
        entryDate   = try c.decode(String.self, forKey: .entryDate)
        notes       = try? c.decode(String.self, forKey: .notes)
        createdAt   = try? c.decode(String.self, forKey: .createdAt)
        updatedAt   = try? c.decode(String.self, forKey: .updatedAt)

        showroomName = (try? c.decode(NameNested.self, forKey: .showroom))?.name
        userName     = (try? c.decode(NameNested.self, forKey: .user))?.name

        let amtRaw = try c.decode(AnyCodable.self, forKey: .cashAmount)
        cashAmount = Double("\(amtRaw.value)") ?? 0

        if let b = try? c.decode(Bool.self, forKey: .isLocked) {
            isLocked = b
        } else {
            isLocked = (try? c.decode(Int.self, forKey: .isLocked)) == 1
        }

        let typeRaw  = (try? c.decode(String.self, forKey: .cashAccountType)) ?? "main"
        cashAccountType  = typeRaw
        cashAccountLabel = (try? c.decode(String.self, forKey: .cashAccountLabel))
            ?? (typeRaw == "mano" ? "Mano's Account" : "Main Account")
    }
}
